logging/src/main/java/org/jboss/as/logging/handlers/AsyncHandlerResourceDefinition.java

```diff
@@ -89,7 +89,7 @@ public void marshallAsElement(final AttributeDefinition attribute, final ModelNo
             .setDefaultValue(new ModelNode(OverflowAction.BLOCK.name()))
             .setPropertyName("overflowAction")
             .setResolver(OverflowActionResolver.INSTANCE)
-            .setValidator(EnumValidator.create(OverflowAction.class, false, false))
+            .setValidator(EnumValidator.create(OverflowAction.class))
```

Comment:
there was a mismatch here. The validator was set with `allow expressions set to false` while the builder itself was using `setAllowExpression(true)`.

AIUI, in the validation code, that means that if there was an expression, it was replaced *before* validation with the validator.

Reply:
Hi, I think the `setAllowExpression(true)` is correct because the `validateParameter()` method in the `EnumValidator` class doesn't seem to check whether the value is true or false.

https://github.com/wildfly/wildfly-core/blob/main/controller/src/main/java/org/jboss/as/controller/operations/validation/EnumValidator.java#L220

Reply:
@jmesnil @clara0 The reason it's ok to just remove this boolean parameters passed to validator constructors, without worrying about their values is that the validators passed to any .setValidator is always wrapped with an internal validator that handles the nullable and expression allowed checks based on the AttributeDefinition settings.

The validator actually used by an AttributeDefinition is always passed through this:

https://github.com/wildfly/wildfly-core/blob/main/controller/src/main/java/org/jboss/as/controller/AttributeDefinition.java#L111

This is to ensure the settings on the AD itself control without inconsistent settings on a validator resulting in different behavior from what the AD describes in :read-resource-description or read-operation-description.

The EnumValidator constructors that take those params are deprecated because the params are cruft due to the above.